src/postgres-util/src/lib.rs

```diff
@@ -284,10 +287,15 @@ impl Config {
                 Ok(client)
             }
             SshTunnelConfig::Tunnel { .. } => {
+                let (session, local_port) = self.establish_openssh_connection().await?;
                 let tls = MakeTlsConnect::<TokioTcpStream>::make_tls_connect(&mut tls, &self.host)?;
-                let channel = self.establish_ssh_connection().await?;
-                let (client, connection) = self.postgres_config.connect_raw(channel, tls).await?;
-                task::spawn(|| connection_task_name, connection);
+                let tcp_stream = TokioTcpStream::connect(("localhost", local_port)).await?;
+                let (client, connection) =
+                    self.postgres_config.connect_raw(tcp_stream, tls).await?;
+                task::spawn(|| connection_task_name, async {
+                    _ = connection.await;
+                    _ = session.close().await;
```

Comment:
I would just do a quick and dirty `tracing::error!` if either of these error! (but always attempt both even if one errors!